app/assets/javascripts/users.js

```diff
@@ -0,0 +1,22 @@
+$(document).ready(function(){
+
+  //set bars for donations by causes. 
+  (function(){
```

Comment:
oh wait i should have been more clear - 
`(function(){` should be first thing in the file

Reply:
I don't understand what you mean. I converted the function to iife.

Reply:
so the purpose of iife when you write things in javascript any thing you declare outside of a function is automatically attached to the global object, which is `window`. 

iife is used so that when you declare functions and variables, they are not exposed.

It doesn't make sense to wrap an iife inside `$(document).ready(function(){` because that's an whole different listener trigger that is called by the `onload` event

Reply:
I'm still not sure you're suggesting. I don't think anything is being exposed the way I have it now. I haven't used javascript in a while, but with jquery I usually just put things in  `$(document).ready(function() { // some code});` or the alternate form `$(function(){
   // some code
});`

Do you have a specific example of what your're talking about?

Reply:
if you're still unclear on this, this was what I requested when I mean to "wrap all of this": 
```
(function() {
	$(document).ready(function(){
	  setDonationBars();
	});


	 //experimental function. just place holder for now. will be properly implemented later. 
	function setDonationBars(){
	  var donations_bars = $(".donations-bars");
	  grid_template_columns = "";

	   $(".donations-bars > .donation-bar").each(function(index){
	    _that = $(this);
	    grid_template_columns = grid_template_columns + " " +  _that.data("percent") + "%";
	  });

	   $(".donations-bars").css("grid-template-columns", grid_template_columns);

	   // this is a hack to adjust space because column grid space pushes div outside of container. 
	  // Will change this during proper implementation.
	  $(".donations-bars").css("padding-right", (($(".donations-bars > div").length -1)* 1) + "px");  
	} 
})();
```

Reply:
if you don't have the wrap of iife your `setDonationBars` function automatically gets tied to the `window` object, which was my reference to "polluting the global space"

Reply:
it's a standard practice in javascript if you're not using transpiling frameworks like webpack.

Reply:
here is another decent blog post about it if you're interested: https://codeburst.io/javascript-what-the-heck-is-an-immediately-invoked-function-expression-a0ed32b66c18

Reply:
> if you don't have the wrap of iife your `setDonationBars` function automatically gets tied to the `window` object, which was my reference to "polluting the global space"

ah, alright I see what you mean now. JS is confusing, but I think it makes more sense now.  I think I'll just stick with ruby.

Reply:
noooo don't leave mee 😭

Reply:
bye :wave: